TLM/TLM/UI/TrafficManagerTool.cs

```diff
@@ -1150,37 +1181,37 @@ void DefaultRenderOverlay(RenderManager.CameraInfo cameraInfo)
                         // segmentInput.m_ignoreSegmentFlags = NetSegment.Flags.Untouchable;
 
                         if (RayCast(segmentInput, out segmentOutput)) {
-                            HoveredSegmentId = segmentOutput.m_netSegment;
+                            segmentId = segmentOutput.m_netSegment;
                         }
                     }
                 }
 
-                if(HoveredSegmentId != 0) {
+                if(segmentId != 0 && raycastSegment) {
                     HitPos = segmentOutput.m_hitPos;
+                    HoveredSegmentId = segmentId;
                 }
 
-                if (HoveredNodeId <= 0 && HoveredSegmentId > 0) {
+                if (HoveredNodeId <= 0 && segmentId > 0) {
                     // alternative way to get a node hit: check distance to start and end nodes
                     // of the segment
                     ushort startNodeId = Singleton<NetManager>
-                                         .instance.m_segments.m_buffer[HoveredSegmentId]
-                                         .m_startNode;
+                                         .instance.m_segments.m_buffer[segmentId].m_startNode;
                     ushort endNodeId = Singleton<NetManager>
-                                       .instance.m_segments.m_buffer[HoveredSegmentId].m_endNode;
+                                       .instance.m_segments.m_buffer[segmentId].m_endNode;
 
                     NetNode[] nodesBuffer = Singleton<NetManager>.instance.m_nodes.m_buffer;
                     float startDist = (segmentOutput.m_hitPos - nodesBuffer[startNodeId]
                                                                 .m_position).magnitude;
                     float endDist = (segmentOutput.m_hitPos - nodesBuffer[endNodeId]
                                                               .m_position).magnitude;
-                    if (startDist < endDist && startDist < 75f) {
+                    if (startDist < endDist && startDist < 15f) {
```

Comment:
magic number :D
does it have a name?

Reply:
I love magic numbers :D I'll figure out a name for it😉